src/blas/backends/cublas/cublas_level3.cpp

```diff
@@ -0,0 +1,383 @@
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  For your convenience, a copy of the License has been included in this
+*  repository.
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+**************************************************************************/
+#include <CL/sycl/detail/pi.hpp>
+#include "cublas_helper.hpp"
+#include "cublas_scope_handle.hpp"
+#include "onemkl/blas/detail/cublas/onemkl_blas_cublas.hpp"
+
+namespace onemkl {
+namespace cublas {
+template <typename Func, typename T>
+inline void gemm(Func func, cl::sycl::queue &queue, transpose transa, transpose transb, int64_t m,
+                 int64_t n, int64_t k, T alpha, cl::sycl::buffer<T, 1> &a, int64_t lda,
+                 cl::sycl::buffer<T, 1> &b, int64_t ldb, T beta, cl::sycl::buffer<T, 1> &c,
+                 int64_t ldc) {
+    using cuDataType = typename CudaEquivalentType<T>::Type;
+    overflow_check(m, n, k, lda, ldb, ldc);
+    queue.submit([&](cl::sycl::handler &cgh) {
+        auto a_acc = a.template get_access<cl::sycl::access::mode::read>(cgh);
+        auto b_acc = b.template get_access<cl::sycl::access::mode::read>(cgh);
+        auto c_acc = c.template get_access<cl::sycl::access::mode::read_write>(cgh);
+        cgh.interop_task([=](cl::sycl::interop_handler ih) {
+            auto sc     = CublasScopedContextHandler(queue);
+            auto handle = sc.get_handle(queue);
+            auto a_     = sc.get_mem<cuDataType *>(ih, a_acc);
+            auto b_     = sc.get_mem<cuDataType *>(ih, b_acc);
+            auto c_     = sc.get_mem<cuDataType *>(ih, c_acc);
+            cublasStatus_t err;
+            CUBLAS_ERROR_FUNC(func, err, handle, get_cublas_operation(transa),
+                              get_cublas_operation(transb), m, n, k, (cuDataType *)&alpha, a_, lda,
+                              b_, ldb, (cuDataType *)&beta, c_, ldc);
+        });
+    });
+}
+
+#define GEMM_LAUNCHER(TYPE, CUBLAS_ROUTINE)                                                        \
+    void gemm(cl::sycl::queue &queue, transpose transa, transpose transb, int64_t m, int64_t n,    \
+              int64_t k, TYPE alpha, cl::sycl::buffer<TYPE, 1> &a, int64_t lda,                    \
+              cl::sycl::buffer<TYPE, 1> &b, int64_t ldb, TYPE beta, cl::sycl::buffer<TYPE, 1> &c,  \
+              int64_t ldc) {                                                                       \
+        gemm(CUBLAS_ROUTINE, queue, transa, transb, m, n, k, alpha, a, lda, b, ldb, beta, c, ldc); \
+    }
+
+GEMM_LAUNCHER(float, cublasSgemm)
+GEMM_LAUNCHER(double, cublasDgemm)
+// GEMM_LAUNCHER(std::complex<float>, cublasCgemm3m) from sm5 onward can improve
```

Comment:
What does "from sm5 onward" mean?

Reply:
It is removed since the function mentioned in the comment was part of BLAS extension and not related to this PR.